src/Features/Core/Portable/Diagnostics/ICodeAnalysisDiagnosticAnalyzerService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Service to compute and fetch analyzer diagnostics from explicit invocation of code analysis on a project or a solution.
+    /// </summary>
+    internal interface ICodeAnalysisDiagnosticAnalyzerService : IWorkspaceService
+    {
+        /// <summary>
+        /// Runs all the applicable analyzers on the given project or entire solution if <paramref name="projectId"/> is null.
+        /// </summary>
+        Task RunAnalysisAsync(Solution solution, Action<Project> onProjectAnalyzed, ProjectId? projectId, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Returns true if <see cref="RunAnalysisAsync(Solution, Action{Project}, ProjectId?, CancellationToken)"/> was invoked
+        /// on either the current or a prior snapshot of the project or containing solution for the given <paramref name="projectId"/>.
+        /// </summary>
+        bool HasProjectBeenAnalyzed(ProjectId projectId);
```

Comment:
doc that once a project is analyzed this always returns true, until the point the workspace's solution is ...